src/services/swap_service.ts

```diff
@@ -798,4 +889,63 @@ export class SwapService implements ISwapService {
 
         return (await this._altRfqMarketsCache.getResultAsync()).result;
     }
+
+    /**
+     * Get sell token fee and corresponding on-chain transfers.
+     *
+     * @param sellToken Sell token address.
+     * @param sellAmount Sell amount.
+     * @param sellTokenAmountPerWei Sell token token amount per 1 wei native token.
+     * @param gasPrice Gas price.
+     * @param quoteGasEstimate Gas estimate for swap quote.
+     * @param feeConfigs Fee configs object.
+     * @param metaTransactionVersion Version of meta-transaction.
+     * @returns Fees object, total sell token amount to charge, sell token on-chain transfer and the gas
+     *          used for on-chain transfers.
+     */
+    private _getSellTokenFees(
+        sellToken: string,
+        sellAmount: BigNumber,
+        sellTokenAmountPerWei: BigNumber | undefined,
+        gasPrice: BigNumber,
+        quoteGasEstimate: BigNumber,
+        feeConfigs: FeeConfigs | undefined,
+        onChainFeeTransferType: 'affiliateFee' | 'transferFrom',
```

Comment:
hmm, it's unclear what `affiliateFee` and `transferFrom` mean in this context.

Reply:
Yeah. `onChainFeeTransferType` here means which way we're going to transfer the fee on-chain. Currently we only have affiliate fee transformer but later in meta-transaction v2 we will use `transferFrom` to directly transfer token instead of using affiliate fee transformer

Reply:
Let me modify the function such the caller would just pass the gas cost for an on-chain transfer in as a parameter. That could simply the logic